core/capabilities/launcher.go

```diff
@@ -216,6 +216,12 @@ func (w *launcher) addRemoteCapabilities(ctx context.Context, myDON registrysync
 					int(remoteDON.F+1),
 					w.lggr,
 				)
+
+				if c.RemoteTriggerConfig == nil {
```

Comment:
@ettec We've added a lot of nil checks here with the same code -- can we somehow move it to a single place that does the initialization?

For example inside remote.NewTriggerPublisher, if it receives a nil config, we instantiate one and call ApplyDefaults on it.

We should apply this pattern for all remote config types too.

Reply:
agree that would be better, moved the check into the publisher/subscriber constructors along with the default application and removed from other places (including common)